### ISSUE TITLE
Retention Policies for Results

### ISSUE THREAD

**Issue by adambkaplan** [tektoncd/results#221]
### Feature request

Allow Tekton Results to define retention policies for parent results and their associated child records.
These retention policies could be defined for the entire cluster, with overrides for specific namespaces.

### Use case

- Data retention compliance.
- Data retention flexibility in multi-tenant environments. If namespaces are used to define tenancy, some namespaces may want results data retained for small periods of time, while others may need data retained long term/permanently.

**Comment by adambkaplan** [comment]
/remove-lifecycle stale

This is still a valid concern.

**Comment by adambkaplan** [comment]
/lifecycle frozen

I agree that having a "v1" that applies to the whole installation makes a lot of sense. I will mark namespace-level features as out of scope so it is easier for contributors to get started.

**Comment by khrm** [comment]
> Would be great to have a cleanup job in the api which would cleanup results and its children records if older than a certain period of time. Maybe we can implement a first version without multi tenant support?

Yes. First,we would implement this and then we are going with filters.

**Comment by enarha** [comment]
@khrm this issue defines two things, global retention policy and per namespace retention policy. We already provide the former. Should we keep issue as a tracker for the latter or do you think it's better to create close this one and create a new issue for the per namespace filtering?
